### PR TITLE
GM: move to platform config

### PR DESCRIPTION
**Pull request by nworb-cire**
**Description**

Followed the example of #31547 and #31550. Also added a `centerToFrontRatio` field since half of the GM cars use 0.5 and half use 0.4.

Note: I'm marking this as WIP until #31550 gets merged, since I merged it into my branch to make use of the `CarSpecs` dataclass.

**Verification**

Pre-commit and other CI testing.

### CODE REVIEW

**Comment by jnewb1** [comment]
any thoughts about the interface?

**Comment by nworb-cire** [comment]
@jnewb1 I think it's good for now. I don't immediately see any fundamental issues with the architecture. Obviously we'll want to move more values into it as time goes on but it's a good start.

File: selfdrive/car/gm/interface.py [hunk 1]
```diff
@@ -165,61 +167,20 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long, docs):
...
-      ret.steerRatio = 15.7
-
     elif candidate == CAR.ACADIA:
       ret.minEnableSpeed = -1.  # engage speed is decided by pcm
-      ret.mass = 4353. * CV.LB_TO_KG
-      ret.wheelbase = 2.86
-      ret.steerRatio = 14.4  # end to end is 13.46
-      ret.centerToFront = ret.wheelbase * 0.4
       ret.steerActuatorDelay = 0.2
       CarInterfaceBase.configure_torque_tune(candidate, ret.lateralTuning)
```
Comment (adeebshihadeh) [inline selfdrive/car/gm/interface.py:173]: most are torque now, want to make another PR to put this on top?

Comment (nworb-cire) [reply selfdrive/car/gm/interface.py:173]: I'm not sure if you're referring to putting another field into the CarSpecs or just making it the default for GM. If you mean the latter, is there a way to unset the torque tune for the couple of cars that don't use it?

Comment (jnewb1) [reply selfdrive/car/gm/interface.py:173]: make it the default for GM. you can reinit the pid tune with
```ret.lateralTuning.init('pid')```

**Review by jnewb1 — APPROVED** [review]

**Comment by jnewb1** [comment]
thanks!
